src/components/GAddressLookup/component.js

```diff
@@ -0,0 +1,271 @@
+                () => this.handleError(),
+            );
+        } else if (!value) {
+            this.clear();
+        }
+    }
+
+    searchGoogleMaps(needle) {
+        const { searchOptions } = this.props;
+        this.setState({ isSearching: true });
+
+        return new Promise((resolve, reject) => {
+            this.autocompleteService.getPlacePredictions(
+                {
+                    ...searchOptions,
+                    input: needle,
+                },
+                (predictions, status) => {
+                    this.setState({ isSearching: false });
+
+                    if (status !== window.google.maps.places.PlacesServiceStatus.OK) {
+                        reject(status);
+                    }
+
+                    resolve(predictions);
+                },
+            );
+        });
+    }
+
+    processSearchResults(results) {
+        this.setState({
+            places: results,
+            suggestions: [
+                {
+                    label: (
+                        <span>
+                            Search for: <b>&lsquo;{this.state.searchValue}&rsquo;</b>
+                        </span>
+                    ),
+                    icon: <SearchValueIcon />,
+                },
+            ].concat(results.map(place => formatSuggestionItem(place, true, <LocationItemIcon />))),
+        });
+    }
+
+    handleError() {
+        this.setState({
+            places: [],
+            suggestions: [
+                {
+                    label: (
+                        <span>
+                            Search for: <b>&lsquo;{this.state.searchValue}&rsquo;</b>
+                        </span>
+                    ),
+                    icon: <SearchValueIcon />,
+                },
+            ],
+        });
+    }
+
+    clear() {
+        this.setState({
+            searchValue: '',
+            places: [],
+            suggestions: [],
+        });
+    }
+
+    render() {
+        const {
+            style,
+            label,
+            error,
+            placeholder,
+            disabled,
+            tabIndex,
+            required,
+            id,
+            name,
+            hideLabel,
+        } = this.props;
+        const { isSearching, suggestions } = this.state;
+        return (
+            <div id={id} style={style} className={this.getContainerClassName()}>
+                <Lookup
+                    id={this.lookupId}
+                    name={name}
+                    label={label}
+                    debounce
+                    isLoading={isSearching}
+                    placeholder={placeholder}
+                    options={suggestions.length > 0 ? suggestions : null}
+                    hideLabel={hideLabel}
+                    required={required}
+                    tabIndex={tabIndex}
+                    disabled={disabled}
+                    value={this.state.selectedPlace}
```

Comment:
here use:
```value={getFormattedValue(value)}```
where value comes from props and getFormattedValue return the option with the label, description and icon computed